src/modules/asset-essentials-rt/asset-essentials-rt.service.ts

```diff
@@ -0,0 +1,98 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { AssetEssentialsDto, AssetEssentialsRealTimeEntity} from 'lib-typeorm';
+
+
+@Injectable()
+export class AssetEssentialsRtService {
+  constructor(
+    @InjectRepository(AssetEssentialsRealTimeEntity)
+    private readonly assetEssentialsRealTimeRepository: Repository<AssetEssentialsRealTimeEntity>
+  ) {}
+
+  
+  async createRealTime(dto: AssetEssentialsDto): Promise<AssetEssentialsRealTimeEntity> {
+    const entity = this.assetEssentialsRealTimeRepository.create(dto);
+    return this.assetEssentialsRealTimeRepository.save(entity);
+  }
+
+  async findAllRealTime(query: any): Promise<AssetEssentialsRealTimeEntity[]> {
+    const queryBuilder = this.assetEssentialsRealTimeRepository.createQueryBuilder('asset_essentials_rt_rt');
+    
+    if (query.isin) {
+      queryBuilder.andWhere('asset_essentials_rt.isin = :isin', { isin: query.isin });
+    }
+
+    if (query.symbol) {
+      queryBuilder.andWhere('asset_essentials_rt.symbol LIKE :symbol', { symbol: `%${query.symbol}%` });
+    }
+
+    if (query.price) {
+      queryBuilder.andWhere('asset_essentials_rt.price = :price', { price: query.price });
+    }
+
+    if (query.high) {
+      queryBuilder.andWhere('asset_essentials_rt.high = :high', { high: query.high });
+    }
+
+    if (query.low) {
+      queryBuilder.andWhere('asset_essentials_rt.low = :low', { low: query.low });
+    }
+
+    if (query.annualChangePercent) {
+      queryBuilder.andWhere('asset_essentials_rt.annualChangePercent = :annualChangePercent', { annualChangePercent: query.annualChangePercent });
+    }
+
+    if (query.yearToDateChangePercent) {
+      queryBuilder.andWhere('asset_essentials_rt.yearToDateChangePercent = :yearToDateChangePercent', { yearToDateChangePercent: query.yearToDateChangePercent });
+    }
+
+    if (query.volume) {
+      queryBuilder.andWhere('asset_essentials_rt.volume = :volume', { volume: query.volume });
+    }
+
+    if (query.changePercent) {
+      queryBuilder.andWhere('asset_essentials_rt.changePercent = :changePercent', { changePercent: query.changePercent });
+    }
+
+    if (query.marketCap) {
+      query.Builder.andWhere('asset_essentials_rt.marketCap = :marketCap', {marketCap: query.marketCap})
+    }
```

Comment:
same as below. WHy we are using query builder

Reply:
Used an alternative method now. Now i am using an array that stores the conditions for the queries